app/src/renderer/vuex/modules/user.js

```diff
@@ -106,27 +112,36 @@ export default ({}) => {
       dispatch(`initializeWallet`, address)
       return address
     },
-    async signIn({ state, commit, dispatch }, { account }) {
-      state.account = account
-      state.signedIn = true
-
-      let keys = await state.externals.loadKeys()
-      let { address } = keys.find(({ name }) => name === account)
-
-      state.address = address
-
+    async signIn(
+      { state, commit, dispatch },
+      { account, sessionType, address }
+    ) {
+      let accountAddress
+      switch (sessionType) {
+        case `ledger`:
+          accountAddress = address
+          break
+        default:
+          // local keyStore
+          state.account = account // TODO: why do we have state.account and state.accounts ??
```

Comment:
I think this should fit into a mutator.

Good question, account + accounts is a bit misleading, It seems accounts comes from the genesis, where instead account is the one you are authenticated with, is that right?

Do we really need to have all the accounts loaded? I don't know how much/fast this list will grow so initially won't be a problem probably

Reply:
I think accounts are all accounts in the keystore, so just yours.

Reply:
we should only have `accounts` then, account could be a `getter`

Reply:
let's move this to another issue if required